docs/website/docs/developers/general/contributing.md

```diff
@@ -2,21 +2,134 @@
 
-<!-- TODO(scotttodd): Update this document
-    * pull more text into this, update that to point to the website
-    * document access controls (join organization then team)
-    * document revert policy
-    * document where new community members should start
--->
+Please [file issues](https://github.com/openxla/iree/issues/new/choose) or
+reach out on any of our other
+[communication channels](../../index.md#communication-channels) before doing
+substantial work; this will ensure that others don't duplicate the work and
+that there's a chance to discuss any design issues.
 
-## Build systems
+## Developer policies
+
+### :octicons-code-of-conduct-16: Code of conduct
+
+This project follows the
+[OpenXLA Code of Conduct](https://github.com/openxla/community/blob/main/CODE-OF-CONDUCT.md).
+
+### :octicons-law-16: Contributor License Agreement
+
+Contributions to this project must be accompanied by a Contributor License
+Agreement (CLA). Head over to <https://cla.developers.google.com/> to see
+your current agreements on file or to sign a new one.
+
+* You (or your employer) retain the copyright to your contribution; this simply
+  gives us permission to use and redistribute your contributions as part of the
+  project.
+* You generally only need to submit a CLA once, so if you've already submitted
+  one (even if it was for a different project), you probably don't need to do it
+  again.
+
+### :octicons-pencil-16: Coding style guidelines
+
+Most of the code style is derived from the
+[Google Style Guides](http://google.github.io/styleguide/) for the appropriate
+language and is generally not something we accept changes on (as clang-format
+and other linters set that for us). The C++ compiler portion of the project
+follows the
+[MLIR/LLVM style guide](https://mlir.llvm.org/getting_started/DeveloperGuide/#style-guide).
+
+Improvements to code structure and clarity are welcome but please file issues
+to track such work first. Pure style changes are unlikely to be accepted unless
+they are applied consistently across the project.
+
+??? tip - "Tip - code formatters and lint scripts"
+
+    Formatters like
+    [`clang-format`](https://clang.llvm.org/docs/ClangFormat.html) (C/C++) and
+    [_Black_](https://black.readthedocs.io/en/stable/) (Python) can be set to
+    run automatically in your editor of choice.
+
+    The script at
+    [`build_tools/scripts/lint.sh`](https://github.com/openxla/iree/blob/main/build_tools/scripts/lint.sh)
+    can also be used to run the full suite of lint checks.
+
+### :octicons-code-review-16: Code reviews
+
+All submissions, including submissions by maintainers, require review. We
+use GitHub pull requests (PRs) for this purpose. Consult
+[GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
+information on using pull requests.
+
+* Please keep PRs small (focused on a single issue) to make reviews and later
+  culprit-finding easier.
```

Comment:
Please add:

`* You may see trusted core contributors bending this rule for project maintenance and major subsystem renovation. If you feel like the rules aren't working for a certain situation, please ask as we bias towards pragmatism for cases that require it.`